### DISCUSSION TITLE
Some ideas

### DISCUSSION THREAD

**Discussion by fouwaru** [jakobdylanc/llmcord.py#14] [Ideas]
I have some Ideas that could be implemented.

- Maybe a feature for the local-llm that the Bot can see wich user is writing and interacting differently
- Maybe better Character Creation.
- Setting to Enable/Disable DM answers
- Ability to Change the status Message

**Comment by jakobdylanc** [comment]
Thank you! :)

- Maybe a feature for the local-llm that the Bot can see wich user is writing and interacting differently
  - (Taken from the new **Notes** section of my README) Currently only OpenAI API supports the **name** property in user messages, therefore only OpenAI API models are user identity aware (with the exception of **gpt-4-vision-preview** which also doesn't support it yet). I tried the alternate approach of prepending user's names in the message content but this didn't seem to work well with all models.
 
- Maybe better Character Creation.
  - Can you elaborate on this more? The CUSTOM_SYSTEM_PROMPT setting in .env lets you set whatever system prompt you want which determines the bot's personality. This should be plenty for creating your own character.
 
- Setting to Enable/Disable DM answers
  - You can actually do this by specifying channel IDs in the ALLOWED_CHANNEL_IDS .env setting. Or specify at least one role in ALLOWED_ROLE_IDS. Otherwise it's pretty simple to disable DMs in the code, just remove `, discord.ChannelType.private` on this line: https://github.com/jakobdylanc/discord-llm-chatbot/blob/7bfaee2564fb17248596c8092ea46e04b4f06608/llmcord.py#L35 Definitely a valid suggestion though. If enough people want a direct setting for disabling DMS then maybe I should add it.
 
- Ability to Change the status Message
  - You can also do this easily by changing `"https://github.com/jakobdylanc/discord-llm-chatbot"` to whatever you want on this line: https://github.com/jakobdylanc/discord-llm-chatbot/blob/7bfaee2564fb17248596c8092ea46e04b4f06608/llmcord.py#L50 (make sure you're on my latest commit or you might not see this)

**Comment by fouwaru** [comment]
So, possibly no way to get that user identity aware for Oobabooga WebUi?

- For Character Creation, maybe an option to use a character.yaml

**Comment by jakobdylanc** [comment]
Oobabooga character support added. Enjoy!
